File: hjkim/Week1/01.05/PGS_프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Integer> queue = new LinkedList<>();
        Queue<Integer> idxQueue = new LinkedList<>();
        for (int i = 0; i < priorities.length; i++) {
            queue.add(priorities[i]);
            idxQueue.add(i);
        }

        while (!queue.isEmpty()) {
            int curr = queue.poll();
            int curr_idx = idxQueue.poll();
            boolean ck = false; // 뒤에 더 큰 우선순위가 있는지 확인
            for (int p : queue) {
                if (curr < p) {
                    ck = true;
                    break;
                }
            }
            if (ck) {
                queue.add(curr);
                idxQueue.add(curr_idx);
            } else {
                answer++;
                if (curr_idx == location) {
                    break;
                }
            }
        }
        return answer;
    }
}

File: raeunlee/Week1/01.05/PGS_프로세스.py
# 정말 이 문제가 왜이리 어려울까

from collections import deque

def solution(priorities, location):
    answer = 1  
    q = deque(priorities)
    idx = location
    
    while q:
        now = q.popleft() # 현재 프로세스
        
        if now < max(q):
            q.append(now)
            if idx == 0:
                idx = len(q) - 1
            else:
                idx -= 1
        else:
            if idx == 0:
                return answer
            else:
                answer += 1
                idx -= 1

    return answer